server/queryconfig.go

```diff
@@ -22,3 +24,28 @@ func ToQueryConfig(query string) chronograf.QueryConfig {
 		Tags: make(map[string][]string, 0),
 	}
 }
+
+var validFieldTypes = map[string]bool{
+	"func":     true,
+	"field":    true,
+	"integer":  true,
+	"number":   true,
+	"regex":    true,
+	"wildcard": true,
+}
+
+// ValidateQueryConfig checks any query config input
+func ValidateQueryConfig(q *chronograf.QueryConfig) error {
+	for _, fld := range q.Fields {
+		invalid := fmt.Errorf(`invalid field type "%s" ; expect func, field, integer, number, regex, wildcard`, fld.Type)
```

Comment:
Hmm. This looks a little weird. Since fields is always has `len > 0` I think maybe `invalid` should come before the `for` loop. This is minor obviously a minor nit.